feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/repository/WalletRepositoryImpl.kt

```diff
@@ -190,33 +204,29 @@ class WalletRepositoryImpl(
         return accountsByAddress[accountAddress]?.name
     }
 
-    private fun createTransaction(hash: String, transfer: Transfer, senderAddress: String, fee: BigDecimal) =
-        Transaction(
+    private fun defineAccountNameForTransaction(
+        accountsByAddress: Map<String, WalletAccount>,
+        displayAddress: String?
+    ): String? {
+        if (displayAddress == null) return null
+        return accountsByAddress[displayAddress]?.name
+    }
+
+    private fun createOperation(hash: String, transfer: Transfer, senderAddress: String, fee: BigDecimal, source: OperationLocal.Source) =
+        OperationLocal(
             hash = hash,
-            tokenType = transfer.tokenType,
-            senderAddress = senderAddress,
-            recipientAddress = transfer.recipient,
-            amount = transfer.amount,
-            date = System.currentTimeMillis(),
-            isIncome = false,
-            fee = fee,
-            status = Transaction.Status.PENDING,
-            accountName = null
+            address = senderAddress,
+            time = System.currentTimeMillis(),
+            tokenType = mapTokenTypeToTokenTypeLocal(transfer.tokenType),
+            call = "Transfer",
```

Comment:
I suggest to extract this constant somewhere, probably to companion object of Transfer Type